mgmt/src/grpc/server.rs

```diff
@@ -0,0 +1,1057 @@
+            }
+        }
+
+        Ok(vpc)
+    }
+
+    /// Convert a gRPC VpcPeering to internal VpcPeering
+    fn convert_peering_from_grpc(
+        &self,
+        peering_grpc: &gateway_config::VpcPeering,
+    ) -> Result<VpcPeering, String> {
+        // Need exactly two VPCs for a peering
+        if peering_grpc.r#for.len() != 2 {
+            return Err(format!(
+                "VPC peering {} must have exactly two VPCs",
+                peering_grpc.name
+            ));
+        }
+
+        // Get the two VPC manifests
+        let vpc1_manifest = self.convert_vpc_manifest_from_grpc(&peering_grpc.r#for[0])?;
+        let vpc2_manifest = self.convert_vpc_manifest_from_grpc(&peering_grpc.r#for[1])?;
+
+        // Create the peering using the constructor
+        Ok(VpcPeering::new(
+            &peering_grpc.name,
+            vpc1_manifest,
+            vpc2_manifest,
+        ))
+    }
+
+    /// Convert gRPC PeeringEntryFor to VpcManifest
+    fn convert_vpc_manifest_from_grpc(
+        &self,
+        entry: &gateway_config::PeeringEntryFor,
+    ) -> Result<VpcManifest, String> {
+        // Create a new VPC manifest with the VPC name
+        let mut manifest = VpcManifest::new(&entry.vpc);
+
+        // Process each expose rule
+        for expose_grpc in &entry.expose {
+            let expose = self.convert_expose_from_grpc(expose_grpc)?;
+            manifest.add_expose(expose).map_err(|e| {
+                format!(
+                    "Failed to add expose to manifest for VPC {}: {}",
+                    entry.vpc, e
+                )
+            })?;
+        }
+
+        Ok(manifest)
+    }
+
+    /// Convert gRPC Expose to VpcExpose
+    fn convert_expose_from_grpc(
+        &self,
+        expose: &gateway_config::Expose,
+    ) -> Result<VpcExpose, String> {
+        // Start with an empty expose
+        let mut vpc_expose = VpcExpose::empty();
+
+        // Process PeeringIP rules
+        for ip in &expose.ips {
+            if let Some(rule) = &ip.rule {
+                match rule {
+                    gateway_config::config::peering_i_ps::Rule::Cidr(cidr) => {
+                        // Parse CIDR into IP and netmask
+                        let (ip_str, netmask) = self.parse_cidr(cidr)?;
+                        // Add as an include rule
+                        vpc_expose = vpc_expose.ip(Prefix::from((ip_str.as_str(), netmask)));
+                    }
+                    gateway_config::config::peering_i_ps::Rule::Not(not) => {
+                        // Parse CIDR into IP and netmask for exclude rule
+                        let (ip_str, netmask) = self.parse_cidr(not)?;
+                        // Add as an exclude rule
+                        vpc_expose = vpc_expose.not(Prefix::from((ip_str.as_str(), netmask)));
+                    }
+                }
+            } else {
+                return Err("PeeringIPs must have either cidr or not field set".to_string());
```

Comment:
Same as the other one - this message might benefit from quotes around `cidr` and `not` for more clarity.

Reply:
fixed :)